### PR TITLE
Updated python flow

### PR DESCRIPTION
**Pull request by Lukewh**
## Done

Changes requested

## QA

- Pull the branch
- Visit http://0.0.0.0:8004/first-snap
- Follow the instructions and ensure everything works and looks correct

### CODE REVIEW

**Comment by bartaz** [comment]
@Lukewh Any details on 'changes requested'? Hard to say what to look at when QAing.

**Comment by therealjuan** [comment]
List of changes for python:
* `Publish your app` call to action on homepage links to first snap flow language selection
* On the homepage `Take 15 minutes to publish your first app` is removed and the title replaces `Try it yourself`
* When selecting python and macOS there is an additional step explaining multipass
* Simplified multipass instructions

image

* The warning above is included in different steps of the first snap flow to make sure it is displayed whenever we specify the snap name

* During the macOS flow we changed Virtual machine from `snaps` to `testvm` to make it clearer

* Included the screenshot for ubuntu one SSO after registering

* After finishing the flow the user is redirected to the listing page of the created snap 
* On the listing page (you can test by adding ?from=fsf to any listing page) there is a notification stating:

```
 **Congratulations** {name} is pushed to the Store. Make it available for everyone or modify its public information here. 
```

You can follow changes in light yellow using the [copydoc](https://docs.google.com/spreadsheets/d/1gN_xcaAFFP_ckDvWeNDpD9ogbRXFgcXN0w0ITZQB0e4/edit#gid=465155665)

**Comment by bartaz** [comment]
In some cases we show examples that use `offlineimap` as a name and then show warning about changing it:

screen shot 2018-11-07 at 16 09 20

but in other cases we show `test-offlineimap-{name}` in the examples and still show the same warning about changing `offlineimap`:

screen shot 2018-11-07 at 16 09 39
screen shot 2018-11-07 at 16 09 28

It's inconsistent and bit confusing.

File: templates/first-snap/build.html [hunk 1]
```diff
@@ -27,6 +27,14 @@ <h4 class="p-list-step__title">
               {% endif %}
             </li>
           {% endfor %}
+          {% if os.startswith('linux') %}
```
Comment (bartaz) [inline templates/first-snap/build.html]: It feels a bit weird to be hardcoding this warnings in the template while they depend on the content (existence of 'offlineimap' in commands).  If the content changes we may end up with showing unnecessary warnings (or not showing warnings when needed). We should find another way to do it.

Maybe there should be some field in content .yaml to turn the warning on? Or the content could be written in a way that avoids the need for them?

File: templates/first-snap/test.html [hunk 2]
```diff
@@ -27,6 +27,14 @@ <h4 class="p-list-step__title">
               {% endif %}
             </li>
           {% endfor %}
+          {% if os.startswith('macos') %}
```
Comment (bartaz) [inline templates/first-snap/test.html]: Same comment as above - conditional warning that depends on the content.

File: templates/publisher/_header.html [hunk 3]
```diff
@@ -1,6 +1,16 @@
 {% include "publisher/_beta-notification.html" %}
 
 <section class="p-strip is-shallow u-no-padding--bottom">
+  {% if from == "fsf" %}
```
Comment (bartaz) [inline templates/publisher/_header.html]: Could it be `?from=first-snap`? It would be more readable. 'fsf' is dangerously close to FFS ;)

**Comment by therealjuan** [comment]
Just did a quick review with @evandandrea on the journey and some minor amends:

* **Test & Push** Update to the warning: Make sure the name matches what has been used previously (i.e test-offlineimap-{name})

image

* **Push:** Update copy alongside button > Once you've pushed to the Snap store, edit your snap public presentation.
image


* **Confirmation:** Update to the confirmation copy:
image

**Comment by Lukewh** [comment]
Todo

- [ ] Add warning flag to `.yaml` files
- [ ] `?from=fsf` -> `?from=first-snap`
- [ ] Test & push, update warning
- [ ] Push update copy
- [ ] Update confirmation copy

**Comment by evandandrea** [comment]
Hold fire on publishing this.

```
$ snapcraft 
Hit http://security.ubuntu.com/ubuntu xenial-security InRelease                                                                                              
Hit http://archive.ubuntu.com/ubuntu xenial InRelease                                                                                                        
Hit http://archive.ubuntu.com/ubuntu xenial-updates InRelease                                                                                                
Hit http://archive.ubuntu.com/ubuntu xenial-backports InRelease                                                                                              
Fetched 0 B in 0s (0 B/s)                                                                                                                                    
Pulling setup 
Fetching and installing pip...
Collecting pip
  Using cached https://files.pythonhosted.org/packages/c2/d7/90f34cb0d83a6c5631cf71dfe64cc1054598c843a92b400e55675cc2ac37/pip-18.1-py2.py3-none-any.whl
  Saved ./parts/setup/python-packages/pip-18.1-py2.py3-none-any.whl
Successfully downloaded pip
Ignoring indexes: https://pypi.python.org/simple
Collecting pip
Installing collected packages: pip
Successfully installed pip-8.1.1
Fetching and installing wheel...
Collecting wheel
  Using cached https://files.pythonhosted.org/packages/5a/9b/6aebe9e2636d35d1a93772fa644c828303e1d5d124e8a88f156f42ac4b87/wheel-0.32.2-py2.py3-none-any.whl
  Saved ./parts/setup/python-packages/wheel-0.32.2-py2.py3-none-any.whl
Successfully downloaded wheel
Looking in links: /home/ubuntu/offlineimap/parts/setup/python-packages
Collecting wheel
Installing collected packages: wheel
Successfully installed wheel-0.32.2
Fetching and installing setuptools...
Collecting setuptools
  Using cached https://files.pythonhosted.org/packages/82/a1/ba6fb41367b375f5cb653d1317d8ca263c636cff6566e2da1b0da716069d/setuptools-40.5.0-py2.py3-none-any.whl
  Saved ./parts/setup/python-packages/setuptools-40.5.0-py2.py3-none-any.whl
Successfully downloaded setuptools
Looking in links: /home/ubuntu/offlineimap/parts/setup/python-packages
Collecting setuptools
Installing collected packages: setuptools
Successfully installed setuptools-40.5.0
Processing /home/ubuntu/offlineimap/parts/setup/src
  Link is a directory, ignoring download_dir
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-req-build-jz5wGs/setup.py", line 26, in <module>
        import offlineimap
      File "offlineimap/__init__.py", line 20, in <module>
        from offlineimap.init import OfflineImap
      File "offlineimap/init.py", line 34, in <module>
        from offlineimap import threadutil, accounts, folder, mbnames
      File "offlineimap/accounts.py", line 23, in <module>
        import six
    ImportError: No module named six
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-req-build-jz5wGs/
Failed to run '/home/ubuntu/offlineimap/parts/setup/install/usr/bin/python2 -m pip download --disable-pip-version-check --dest /home/ubuntu/offlineimap/parts/setup/python-packages six .': Exited with code 1.
```

**Comment by evandandrea** [comment]
```
The python plugin builds upon the work you’ve already done to describe your Python dependencies in setup.py and requirements.txt. It will automatically include these in your snap.
```

We discovered that this is wrong. Let's change it to (and updating the spreadsheet at the same time):
```
The python plugin builds upon the work you’ve already done to describe your Python dependencies in setup.py and requirements.txt. You can include the latter by setting the `requirements` keyword to the path of your requirements.txt.
```

File: templates/first-snap/build.html [hunk 4]
```diff
@@ -11,21 +11,33 @@
...
-              {% endif %}
-            </li>
+              </li>
+            {% endif %}
+          {% endfor %}
+          {% for step in steps %}
+            {% if loop.last and step.warning %}
+              <div class="p-notification--caution">
+                <p class="p-notification__response">
+                  <span class="p-notification__status"></span>
```
Comment (bartaz) [inline templates/first-snap/build.html]: Is it needed if empty?

File: templates/first-snap/test.html (hunk 2 above)
Comment (bartaz) [reply templates/first-snap/test.html]: Could you use same `{{ step.warning|safe }}` as in linux case?

**Comment by bartaz** [comment]
'Waiting for push' button has hover state and cursor even that it cannot be clicked.

screen shot 2018-11-09 at 17 05 39

Could we make it disabled?

**Comment by evandandrea** [comment]
@Lukewh I've resolved the issue with the python example snapcraft.yaml failing to build. I've pushed that to https://github.com/snapcraft-docs/offlineimap and updated the first snap flow copydoc.

**Pull request merged**